cygnus-twitter/README.md

```diff
@@ -1 +1,296 @@
+
+Current stable release is able to persist Twitter data in:
+
+* [HDFS](http://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/HdfsUserGuide.html), the [Hadoop](http://hadoop.apache.org/) distributed file system.
+
+You may consider to visit [Cygnus-twitter User and Programmer Guide](doc/cygnus-twitter/user_and_programmer_guide/README.md) before going deep into the details.
+
+[Top](#top)
+
+##<a name="section2"></a>Basic operation
+###<a name="section2.1"></a>Hardware requirements
+* RAM: 1 GB, specially if abusing of the batching mechanism.
+* HDD: A few GB may be enough unless the channel types are configured as `FileChannel` type.
+
+[Top](#top)
+
+
+###<a name="section2.3"></a>Configuration
+Cygnus-twitter is a tool with a high degree of configuration required for properly running it.
+
+So, the starting point is choosing the internal architecture of the Cygnus agent. Let's assume the simplest one:
+
+```
++----------------+    +----------------+    +----------------+
+| Twitter source |----| memory channel |----|  Twitter Sink  |
++----------------+    +----------------+    +----------------+
+```
+
+Attending to the above architecture, the content of `/usr/cygnus/conf/cygnus_1.conf` will specify the following parameters:
+
+* The source:
+`cygnus-twitter.sources.http-source.type = org.telefonica.iot.cygnus.sources.TwitterSource`
+
+* The keyworks (hashtags) that are used in the twitter query to filter tweets with an specific keyword(s):
+        
+`cygnus-twitter.sources.twitter-source.keywords = keyword1, keyword2, keyword3`
+
+* The coordinates to specify the spatial area where the source will collect geo-located tweets. The coordinates will be used in the twitter query:
+
+```
+        cygnus-twitter.sources.twitter-source.south-west_latitude = 39.4247692
+        cygnus-twitter.sources.twitter-source.south-west_longitude = -0.4315448
+        cygnus-twitter.sources.twitter-source.north-east_latitude = 39.5038788
+        cygnus-twitter.sources.twitter-source.north-east_longitude = -0.3124204
+```
+
+These coordinates are used to define a rectangle filter where tweets have been geo-located. Only tweets inside this rectangle are stored.
+
+```
+                     -------------- north-east
+                    |                  |
+                    |                  |
+                    |                  |
+               south-west ------------   
+        
+```
+
+ * The credentials used to connect with Twitter API:
+
+```
+        cygnus-twitter.sources.twitter-source.consumerKey = xxxxxxx
```

Comment:
Because these lines are eclosed by ```   ```, the 8 while spaces indentation is not necessary.